### PR TITLE
Add timeout to HTTPX calls

### PR DESCRIPTION
**Pull request by DamianNud**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/25973|

## Description

As commented [here](https://github.com/wazuh/wazuh/issues/25973#issuecomment-2397986246), it was decided to specify a timeout of 10 seconds to each `httpx` call, which sets 5 seconds by default, to be consistent across all calls and with the Management API.

No new definitions were added to the exception catching of each call because `httpx.TimeoutException` inherits from `httpx.RequestError`, which means that the specific timeout error will be caught with its parent class.

There were no changes in the behavior of the calls besides increasing the default timeout value.

### CODE REVIEW

File: framework/wazuh/core/manager.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 WAZUH_UID_KEY = 'wazuh-uid'
 WAZUH_TAG_KEY = 'wazuh-tag'
 USER_AGENT_KEY = 'user-agent'
+DEFAULT_TIMEOUT = 10.0
```
Comment (GGP1) [inline framework/wazuh/core/manager.py:38]: Add a new line to conform with the PEP8 style guide (classes should be separed by two empty lines)

```suggestion
DEFAULT_TIMEOUT = 10.0

```

**Review by GGP1 — CHANGES_REQUESTED** [review]
We could update the unit tests to assert that the `httpx.AsyncClient` mocked instances are being called with the timeouts specified.

This way, we ensure that if they are removed in the future, the tests will fail and we will notice it.

**Comment by DamianNud** [comment]
@GGP1 That is a good idea. I just added test cases checking that `httpx` is called with a timeout.

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by fdalmaup — APPROVED** [review]
LGTM, although the ones related to the HAPHelper could have been omitted and the Bandit alerts dismissed since the component will be removed in 5.0.

**Pull request merged**